ddtrace/tracer/option.go

```diff
@@ -1191,6 +1191,43 @@ func AnalyticsRate(rate float64) StartSpanOption {
 	return Tag(ext.EventSampleRate, rate)
 }
 
+// WithConfig merges the given StartSpanConfig into the one used to start the span.
+// It is useful when you want to set a common base config, reducing the number of function calls in hot loops.
+func WithConfig(cfg ddtrace.StartSpanConfig) StartSpanOption {
+	return func(c *ddtrace.StartSpanConfig) {
+		// copy cfg into c only if cfg fields are not zero values
+		// c fields have precedence, as they may have been set up before running this option
+		if c.SpanID == 0 {
+			c.SpanID = cfg.SpanID
+		}
+
+		if c.Parent == nil {
+			c.Parent = cfg.Parent
+		}
+
+		if c.Context == nil {
+			c.Context = cfg.Context
+		}
+
+		if c.StartTime.IsZero() {
+			c.StartTime = cfg.StartTime
+		}
+
```

Comment:
```suggestion
		if c.SpanID == 0 {
			c.SpanID = cfg.SpanID
		}
		if c.Parent == nil {
			c.Parent = cfg.Parent
		}
		if c.Context == nil {
			c.Context = cfg.Context
		}
		if c.StartTime.IsZero() {
			c.StartTime = cfg.StartTime
		}
```
Suggestion based on the style guidelines https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#general-guidelines